### PR TITLE
added missing classes test for multilabel

### PR DESCRIPTION
**Pull request by aditya1503**
adding missing class labels test for multi-labels
```
pred_probs = np.array(
         [
             [0.9, 0.1, 0.0, 0.4],
             [0.7, 0.8, 0.2, 0.3],
             [0.9, 0.8, 0.4, 0.2],
             [0.1, 0.1, 0.8, 0.3],
             [0.4, 0.5, 0.1, 0.1],
             [0.1, 0.1, 0.2, 0.1],
             [0.8, 0.1, 0.2, 0.1],
         ]
     )
     labels = [[0], [0, 1], [0, 1], [2], [0, 2, 3], [], []]
     noise_idx = filter.find_label_issues(
         labels=labels,
         pred_probs=pred_probs,
         multi_label=True,
         return_indices_ranked_by=return_indices_ranked_by,
     )
```

### CODE REVIEW

File: cleanlab/internal/validation.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def assert_valid_inputs(
         else:
             assert isinstance(y, list)
             assert all(isinstance(y_i, list) for y_i in y)
-            highest_class = max([max(y_i) for y_i in y]) + 1
+            highest_class = max([max(y_i) for y_i in y if len(y_i) != 0]) + 1
```
Comment (aditya1503) [inline cleanlab/internal/validation.py:92]: can't take `max([])`

File: cleanlab/count.py [hunk 2]
```diff
@@ -74,6 +75,8 @@ def num_label_issues(
     labels :
       An array of shape ``(N,)`` of noisy labels, i.e. some labels may be erroneous.
       Elements must be in the set 0, 1, ..., K-1, where K is the number of classes.
+      For multi-label classification where each example can belong to multiple classes,
+      refer to documentation for this argument in filter.find_label_issues() for details.
```
Comment (jwmueller) [inline cleanlab/count.py:79]: need to link to the function properly, see development.md for instructions on how to link to other methods in docstrings

File: cleanlab/count.py [hunk 3]
```diff
@@ -90,6 +93,8 @@ class 0, 1, ..., K-1. `pred_probs` should have been computed using 3 (or
       Entry ``(j, k)`` in the matrix is the number of examples confidently counted into the pair of ``(noisy label=j, true label=k)`` classes.
       The `confident_joint` can be computed using :py:func:`count.compute_confident_joint <cleanlab.count.compute_confident_joint>`.
       If not provided, it is computed from the given (noisy) `labels` and `pred_probs`.
+      If `multi_label` is True, then the `confident_joint` should be a one-vs-rest array of shape ``(K, 2, 2)``, refer to filter.find_label_issues() for detailsf.
```
Comment (jwmueller) [inline cleanlab/count.py:96]: typo and need to link to function properly

File: cleanlab/count.py [hunk 4]
```diff
@@ -105,13 +110,22 @@ class 0, 1, ..., K-1. `pred_probs` should have been computed using 3 (or
 
        TL;DR: use this method to get the most accurate estimate of number of label issues when you don't need the indices of the label issues.
 
+    multi_label : bool, optional
+       Refer to documentation for this argument in ``count.compute_confident_joint()`` with `multi_label=True` for details.
```
Comment (jwmueller) [inline cleanlab/count.py:114]: need to link to function properly

File: cleanlab/filter.py [hunk 5]
```diff
@@ -218,7 +218,18 @@ class 0, 1, ..., K-1.
         "confident_learning",
         "predicted_neq_given",
     ]  # TODO: change default to confident_learning ?
-    assert_valid_inputs(X=None, y=labels, pred_probs=pred_probs, multi_label=multi_label)
+    allow_one_class = False
+    if isinstance(labels, np.ndarray):
+        if set(labels) == {0}:
+            allow_one_class = True
```
Comment (jwmueller) [inline cleanlab/filter.py:224]:
```suggestion
        if set(labels) == {0}:  # occurs with missing classes in multi-label settings
            allow_one_class = True
```

File: cleanlab/internal/validation.py (hunk 1 above)
Comment (jwmueller) [reply cleanlab/internal/validation.py:92]: good catch!

File: cleanlab/rank.py [hunk 6]
```diff
@@ -87,7 +87,17 @@ def order_label_issues(
       ordered by the label-quality scoring method passed to `rank_by`.
     """
 
-    assert_valid_inputs(X=None, y=labels, pred_probs=pred_probs, multi_label=False)
+    allow_one_class = False
+    if isinstance(labels, np.ndarray):
+        if set(labels) == {0}:
```
Comment (jwmueller) [inline cleanlab/rank.py:92]:
```suggestion
        if set(labels) == {0}:  # occurs with missing classes in multi-label settings
```

File: tests/test_filter_count.py [hunk 7]
```diff
@@ -779,6 +779,26 @@ def test_num_label_issues(confident_joint):
...
+        multi_label=True,
+    )
+    f = filter.find_label_issues(
+        labels=dataset["labels"],
+        pred_probs=dataset["pred_probs"],
+        confident_joint=dataset["cj"] if confident_joint else None,
+        multi_label=True,
+    )
+    assert sum(f) == 53
+    assert sum(f) == n
```
Comment (jwmueller) [inline tests/test_filter_count.py:799]:
```suggestion
    assert sum(f) == n
```

Comment (jwmueller) [reply tests/test_filter_count.py:799]: don't think we need to test filter.find_label_issues here too?

File: cleanlab/count.py (hunk 2 above)
Comment (aditya1503) [reply cleanlab/count.py]: Fixed: https://github.com/aditya1503/cleanlab/blob/a684583aa0e89de6414172c891b1db305e31ade3/cleanlab/count.py#L79

File: cleanlab/count.py (hunk 3 above)
Comment (aditya1503) [reply cleanlab/count.py]: Fixed: https://github.com/aditya1503/cleanlab/blob/a684583aa0e89de6414172c891b1db305e31ade3/cleanlab/count.py#L96

File: cleanlab/count.py (hunk 4 above)
Comment (aditya1503) [reply cleanlab/count.py]: Fixed: https://github.com/aditya1503/cleanlab/blob/a684583aa0e89de6414172c891b1db305e31ade3/cleanlab/count.py#L114

File: cleanlab/filter.py (hunk 5 above)
Comment (aditya1503) [reply cleanlab/filter.py]: Fixed: https://github.com/aditya1503/cleanlab/blob/a684583aa0e89de6414172c891b1db305e31ade3/cleanlab/filter.py#L223

File: cleanlab/rank.py (hunk 6 above)
Comment (aditya1503) [reply cleanlab/rank.py]: Fixed: https://github.com/aditya1503/cleanlab/blob/a684583aa0e89de6414172c891b1db305e31ade3/cleanlab/rank.py#L92

File: tests/test_filter_count.py (hunk 7 above)
Comment (aditya1503) [reply tests/test_filter_count.py]: Fixed: https://github.com/aditya1503/cleanlab/blob/a684583aa0e89de6414172c891b1db305e31ade3/tests/test_filter_count.py#L798

**Pull request merged**